internal/operator/operator_group_test.go

```diff
@@ -0,0 +1,46 @@
+package operator
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/opdev/opcap/internal/logger"
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("OperatorGroup", func() {
+	logger.InitLogger("debug")
+	scheme := runtime.NewScheme()
+	operatorv1.AddToScheme(scheme)
+	client := fake.NewClientBuilder().WithScheme(scheme).Build()
+	var operatorClient operatorClient = operatorClient{
+		Client: client,
+	}
+
+	Context("OperatorGroup", func() {
+		It("should exercise OperatorGroup", func() {
+			operatorGroupData := OperatorGroupData{
+				Name:             "testog",
+				TargetNamespaces: []string{"default", "testns"},
+			}
+			By("creating a OperatorGroup", func() {
+				og, err := operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(og).ToNot(BeNil())
+			})
+			By("creating it again should error", func() {
+				og, err := operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError("could not create operatorgroup: testog: operatorgroups.operators.coreos.com \"testog\" already exists"))
+				Expect(og).To(BeNil())
```

Comment:
Hi @yashoza19 , with this fake client you used does it retain the testns in memory after the first By clause and fails on this second one? How does it know that it's creating a duplicated operatorGroup?

Reply:
Hi @acmenezes, in the second test, we are not deleting the operator group created by first test and passing in the same namespace. So it should expect an error `\"testog\" already exists"`. Which is what it is checking for

Reply:
It looks like you're running all those tests in one take. Shouldn't they be separate and for each one you would feed the fake client with fake data according to the specific test?

Reply:
I believe it should be in one take, I'm not sure if the fake client can retain it in memory. That's how the tests were written for the namespace functions.

Reply:
So if is in one take it's not unit test right? Single function. One test depending on the other is never a good thing.

Reply:
The examples I've seen before would setup the fake client with fake data. There is a way to do that. @bcrochet may be the expert around that could clarify this.

Reply:
Ok. I see that @skattoju already approved. I guess that's fine. I'll approve it too.

Reply:
@acmenezes I'll update the tests in a new PR since this one already got merged. Thanks for pointing this out.